### PR TITLE
Rule Migration (Rule ApprovableAware + LocalRuleDAO)

### PR DESCRIPTION
**Pull request by tharmaman**

### CODE REVIEW

File: src/services [hunk 1]
```diff
@@ -724,17 +724,45 @@ p({
...
+    {
+      "of": "foam.nanos.ruler.Rule"
+    }
+  """
+})
+
+p({
+  "class": "foam.nanos.boot.NSpec",
+  "name": "localRuleDAO",
+  "serve": true,
```
Comment (jlhughes) [inline src/services]: local daos should not be served.

**Review by jlhughes — CHANGES_REQUESTED** [review]

**Comment by noodlemoodle** [comment]
lgtm

File: src/services (hunk 1 above)
Comment (tharmaman) [reply src/services]: fixed changed some other local daos to serve false too which were previously serve true

File: src/services [hunk 2]
```diff
@@ -724,17 +724,44 @@ p({
...
   "serviceScript": """
+    dao = new foam.dao.EasyDAO.Builder(x)
+      .setEnableInterfaceDecorators(false)
+      .setOf(foam.nanos.ruler.Rule.getOwnClassInfo())
+      .setRuler(false)
+      .setInnerDAO(x.get("localRuleDAO"))
+      .build();
+
+    dao.addPropertyIndex(new foam.core.PropertyInfo[] { foam.nanos.ruler.Rule.NAME });
+    dao = new foam.dao.ValidatingDAO(x, dao);
```
Comment (lchanmann) [inline src/services:735]: Why not setValidated(true) on the easydao? or maybe not needed as the local one already has it.

File: src/services [hunk 3]
```diff
@@ -724,17 +724,44 @@ p({
...
     dao = new foam.dao.EasyDAO.Builder(x)
       .setPm(true)
       .setGuid(true)
       .setOf(foam.nanos.ruler.Rule.getOwnClassInfo())
       .setJournalType(foam.dao.JournalType.SINGLE_JOURNAL)
       .setJournalName("rules")
+      .setContextualize(true)
+      .setValidated(true)
       .setRuler(false)
+      .setAuthorize(false)
```
Comment (lchanmann) [inline src/services:760]: Why do we not authorize the dao? Imagine, if you give me access to scriptDAO I can get localRuleDAO and I can change any rules.

Comment (kgrgreer) [reply src/services:760]: If we give you access to scriptDAO you can already do whatever you want.

Comment (kgrgreer) [reply src/services:760]: The point of localDAO's is that they're unauthorized to improve performance and data accessibility by the system.

**Pull request merged**